llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -1886,6 +1891,10 @@ SampleProfileLoader::buildFunctionOrder(Module &M, LazyCallGraph &CG) {
     errs() << "WARNING: -use-profiled-call-graph ignored, should be used "
               "together with -sample-profile-top-down-load.\n";
 
+  for (Function &F : M)
+    if (!F.isDeclaration() && BitCodeUseSampleProfile)
+      F.addFnAttr("use-sample-profile");
```

Comment:
This is not the starting place the attribute will be used. `MatchingManager->runOnModule()` also needs to see the attribute.